### PR TITLE
pretty error report's output

### LINKED ISSUE #38: improve error & code output
- [x] use [pretty printer](https://idris2docs.sinyax.net/data/contrib/docs/Text.PrettyPrint.Prettyprinter.Render.Terminal.html#Text.PrettyPrint.Prettyprinter.Render.Terminal.AnsiStyle)
- [x] improve parse error output
- [x] improve check error output

### PR DESCRIPTION
**Pull request by dannypsnl**
NOTE
- limit parsing error to first error
  - since idris's parser reports too many errors(some seems repeated)
  - this is a trade off decision
- pretty parsing error
- pretty type checking error
- pretty core term

resolve #38

### CODE REVIEW

File: src/Violet/Parser.idr [hunk 1]
```diff
@@ -108,21 +110,22 @@ mutual
     u <- tm
     pure $ RLet name a t u
 
-parseTokens : List (WithBounds VToken) -> Either String Raw
-parseTokens toks =
+parseTokens : List (WithBounds VToken) -> (ParsingError VToken -> PError) -> Either (Doc AnsiStyle) Raw
```
Comment (rayshih) [inline src/Violet/Parser.idr:113]: I'm not sure if this is a concern: if we bind the Ansi annotation deeply with parsing process, this might become an issue if we want to integrate this with LSP

Comment (dannypsnl) [reply src/Violet/Parser.idr:113]: Yes, this would be a problem. But since this PR will block your work, so I tend to solve this in another PR

Comment (dannypsnl) [reply src/Violet/Parser.idr]: Fix it, postpone it to the Main

File: src/Violet/Error/Check.idr [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import Text.Parser.Core
+
+import Violet.Error.Common
+
+public export
+record CheckError ann where
+  constructor MkCheckError
+  filename, source : String
+  bounds : Maybe Bounds
+  msg : Doc ann
```
Comment (dannypsnl) [inline src/Violet/Error/Check.idr:12]: @rayshih This is a problem too, I would make another PR to remove this render dependencies. Here the checker will need a definition like
```idris
data CheckErrorKind
  = NoVar Name
  | TypeMismatching Tm Tm
  | ...
```

File: src/Violet/Error/Parsing.idr [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import Violet.Error.Common
+import Violet.Lexer
+
+repeat : Nat -> Doc ann -> List (Doc ann)
+repeat n s = iterateN n (\x => x) s
+
+export
+record PError where
+  constructor MkPError
+  source : String
```
Comment (dannypsnl) [inline src/Violet/Error/Parsing.idr:15]: Also another PR, I think `filename : String` needed here
